core/tests/move_file_tests.rs

```diff
@@ -259,4 +259,103 @@ mod move_document_tests {
             ))
         );
     }
+
+    #[test]
+    fn move_folder_cannot_move_root() {
+        // new account
+        let account = generate_account();
+        let folder_id = Uuid::new_v4();
+        let folder_key = AesImpl::generate_key();
+        let version = ClientImpl::new_account(
+            &account.api_url,
+            &account.username,
+            &sign(&account),
+            account.keys.to_public_key(),
+            folder_id,
+            FolderAccessInfo {
+                folder_id: folder_id,
+                access_key: aes_key(&folder_key, &folder_key),
+            },
+            rsa_key(&account.keys.to_public_key(), &folder_key),
+        )
+        .unwrap();
+
+        // moving root into itself
+        assert_matches!(
+            ClientImpl::move_folder(
+                &account.api_url,
+                &account.username,
+                &sign(&account),
+                folder_id,
+                version,
+                folder_id,
+                FolderAccessInfo {
+                    folder_id: folder_id,
+                    access_key: aes_key(&folder_key, &folder_key),
+                },
+            ),
+            Err(ApiError::<MoveFolderError>::Api(
+                MoveFolderError::CannotMoveRoot
+            ))
+        );
+    }
+
+    #[test]
+    fn move_folder_into_itself() {
```

Comment:
Can you also write a test that makes a folder structure like `root -> a -> b -> c` and then try moving `a` into `c`? I think it will fail